src/Command/Pull/PullCommandBase.php

```diff
@@ -281,7 +281,8 @@ protected function downloadDatabaseDump(
     if ($response->getStatusCode() !== 200) {
       throw new AcquiaCliException("Unable to download database copy from {$url}. {$response->getStatusCode()}: {$response->getReasonPhrase()}");
     }
-    $backup_file = $response->getBody()->getContents();
+    // Get the response body as a stream to avoid ludicrous memory usage.
+    $backup_file = $response->getBody();
```

Comment:
I didn’t realize you could just dump the stream to disk.

Reply:
Yeah I had to check the docs, but sure enough file_put_contents (which the Symfony filesystem uses) [accepts a stream resource](https://www.php.net/manual/en/function.file-put-contents.php), and the Guzzle body also [provides a stream resource](https://docs.guzzlephp.org/en/stable/psr7.html#body), so it all just works.